File: CS/WindowsFormsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using System.Drawing.Design;
namespace WindowsFormsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            XtraReport1 report = new XtraReport1();
            ReportDesignTool tool = new ReportDesignTool(report);
            tool.DesignForm.DesignMdiController.DesignPanelLoaded += new DevExpress.XtraReports.UserDesigner.DesignerLoadedEventHandler(DesignMdiController_DesignPanelLoaded);
            tool.ShowDesignerDialog();
            
        }

        void DesignMdiController_DesignPanelLoaded(object sender, DevExpress.XtraReports.UserDesigner.DesignerLoadedEventArgs e) {
            IToolboxService ts =
    (IToolboxService)e.DesignerHost.GetService(typeof(IToolboxService));
           ts.AddToolboxItem(new ToolboxItem(typeof(CustomControls.MyCompositeControl)), "New Category");

        }
    }
}

File: VB/WindowsFormsApplication1/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports System.Drawing.Design
Namespace WindowsFormsApplication1
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btShowDesigner_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btShowDesigner.Click
			If Not Me.btShowDesigner.IsHandleCreated Then Return

			Dim report As New XtraReport1()
			Dim tool As New ReportDesignTool(report)
			AddHandler tool.DesignForm.DesignMdiController.DesignPanelLoaded, AddressOf DesignMdiController_DesignPanelLoaded
			tool.ShowDesignerDialog()
		End Sub

		Private Sub DesignMdiController_DesignPanelLoaded(ByVal sender As Object, ByVal e As DevExpress.XtraReports.UserDesigner.DesignerLoadedEventArgs)
			Dim ts As IToolboxService = DirectCast(e.DesignerHost.GetService(GetType(IToolboxService)), IToolboxService)
			ts.AddToolboxItem(New ToolboxItem(GetType(CustomControls.MyCompositeControl)), "New Category")

		End Sub


	End Class
End Namespace
